src/lantern/src/Cuda.cpp

```diff
@@ -234,4 +235,26 @@ void* _lantern_cuda_get_rng_state (int device) {
   auto gen = at::detail::getCUDAHooks().getDefaultCUDAGenerator(device);
   return make_raw::Tensor(gen.get_state());
   LANTERN_FUNCTION_END
+}
+
+void _lantern_cuda_record_memory_history(const std::string* enabled, const std::string* context, const std::string& stacks, size_t max_entries) {
```

Comment:
Unfortunatelly we can't have C++ strings across the liblantern and R boundaries (because they might be compiled with different non ABI compatible compilers). You need to take them as `void *` arguments and convert them to proper strings using `from_raw::string()`. See eg: for an example:

https://github.com/mlverse/torch/blob/ca61df853546f6e2248f5de718d0f9d9c7163a8c/src/lantern/src/Compile.cpp#L28-L33